src/main/java/frc/robot/subsystems/AddressableLightStrip.java

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+/*
+ * Asimov's Laws:
+ * The First Law: A robot may not injure a human being or, through inaction, allow a human being to come to harm.
+ * The Second Law: A robot must obey the orders given it by human beings except where such orders would conflict with the First Law.
+ * The Third Law: A robot must protect its own existence as long as such protection does not conflict with the First or Second Law.
+ */
+
+package frc.robot.subsystems;
+
+import edu.wpi.first.networktables.GenericEntry;
+import edu.wpi.first.wpilibj.AddressableLED;
+import edu.wpi.first.wpilibj.AddressableLEDBuffer;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.util.Color;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import org.jspecify.annotations.Nullable;
+
+/** The subsystem for the lights on the robot. */
+public class AddressableLightStrip extends SubsystemBase {
+
+  private final AddressableLED ledStrip;
+  private final AddressableLEDBuffer buffer;
+  private int rainbowHue = 0;
+  private double fadeSpeed = Constants.LightStrips.STRIP_FADE_SPEED;
+  @Nullable private Color targetStripColor = null;
```

Comment:
wait is the annotations lib working?

Reply:
It's been working since I added it 3 months ago.

Reply:
lmao hasn't looked to be working for me but maybe my vscode is just dumb